rfcs/select-component.md

```diff
@@ -0,0 +1,191 @@
+# Select Component
+
+### Exports
+
+* `StatefulSelect`
+* `StatefulSelectContainer`
+* `Select`
+* `StyledSelectInput`
+* `StyledOption`
+* `StyledDropDown`
+* `StyledLabel`
+* `StyledHint`
+
+### `Select` and `StatefulSelect` API
+
+* `type: TYPE.SEARCH | TYPE.SELECT`:
+  type of component to be in select or search mode (with lookup input)
+* `selectedOptions: Array<Object>`:
+  Current selected options. Every option object has `id: string` and `label: React$Node`. Label is defaulted to display for selected option, otherwise see `getSelectedOptionLabel` method
+* `placeholder: ?string`:
+  Placeholder text if nothing is selected. Default is `Choose one...`
+* `rows: ?number`:
+  Represents maximum visible length of options, all other will be scrolled. If not defined, all options will be visible.
+* `options: Array<Object>`:
+  All Options in dropdown. Should be provided for Select and Search mode equally. Every option object has `id: string` and `label: React$Node`. Label is defaulted to display for option in dropdown, otherwise see `getOptionLabel` method. Optional `disabled: boolean` for option to be disabled from selection.
+* `label: ?React$Node`:
+  Component or String representing label for Select component. Default is `''`
+* `caption: ?React$Node`:
+  Component or String representing hint message for Select component. Default is `''`
+* `textValue: ?string`:
+  Initial text value for Search to lookup for text.
```

Comment:
I think `textValue` should be internal component state of `Select`. The vast majority the time I use form controls, I'm using the stateless version, because I need to pass in the current value based on redux or parent component state. We shouldn't force users to create a `textValue` state property in their parent component, the `Select` component should just handle this automatically.

Reply:
in this case it would be impossible to pre-setup input value to some initial state, right?

Reply:
Yup. And I think that's perfectly fine – I've never seen a product where that's a requirement.

If necessary we could have an "initialTextValue" prop, but I don't think it's needed at this point.

Reply:
fixed